include/openssl/evp.h

```diff
@@ -1142,17 +1142,17 @@ OPENSSL_EXPORT EVP_PKEY *EVP_PKEY_new_mac_key(int type, ENGINE *engine,
                                               size_t mac_key_len);
 
 
-// General No-op Functions [Deprecated].
+// Deprecated functions
 
-// EVP_PKEY_get0 returns NULL. This function is provided for compatibility with
-// OpenSSL but does not return anything. Use the typed |EVP_PKEY_get0_*|
-// functions instead.
+// EVP_PKEY_get0 returns the consumed key. The type of value returned differs
+// depending on the type of the |EVP_PKEY|.
 //
-// Note: In OpenSSL, the returned type will be different depending on the type
-//       of |EVP_PKEY| consumed. This leads to misuage very easily and has been
-//       deprecated as a no-op to avoid so.
+// This function is provided only for compatibility with OpenSSL.
+// Prefer the use the typed |EVP_PKEY_get0_*| functions instead.
 OPENSSL_EXPORT OPENSSL_DEPRECATED void *EVP_PKEY_get0(const EVP_PKEY *pkey);
```

Comment:
hm, should we be using `OPENSSL_DEPRECATED` elsewhere? i'm seeing [some nominally deprecated functions](https://github.com/aws/aws-lc/blob/5fd0f085dbc4030b309e6572893f7d5ee074f935/include/openssl/pkcs7.h#L104) in pkcs7.h that don't use this.

Reply:
Our use of `OPENSSL_DEPRECATED` has been discussed a few times. Generally, unless there's a security concern, we're hesitant to add this designation to existing functions as it could potentially break consumers' builds.

In this case, the function was already marked `OPENSSL_DEPRECATED` so I just left it in place.